Sources/Amplitude/AMPStorage.m

```diff
@@ -0,0 +1,107 @@
+//
+//  File.m
+//
+//
+//  Created by Dante Tam on 6/10/21.
+//
+
+#import <Foundation/Foundation.h>
+
+#import "AMPStorage.h"
+
+@interface AMPStorage ()
+@end
+
+@implementation AMPStorage {
+    
+}
+
++ (NSString *)getAppStorageAmpDir {
+    NSArray *paths = NSSearchPathForDirectoriesInDomains(NSApplicationSupportDirectory, NSUserDomainMask, YES);
+    NSString *path = [paths firstObject];
+    NSString *bundleIdentifier = [[NSBundle mainBundle] bundleIdentifier];
+    return [NSString stringWithFormat:@"%@/%@/%@", path, @"/", bundleIdentifier];
+}
+
++ (NSString *)getDefaultEventsFile {
+    NSString *baseDir = [AMPStorage getAppStorageAmpDir];
+    NSString *path = [baseDir stringByAppendingString:@"/amplitude_event_storage.txt"];
+    return path;
+}
+
++ (NSString *)getDefaultIdentifyFile {
+    NSString *baseDir = [AMPStorage getAppStorageAmpDir];
+    NSString *path = [baseDir stringByAppendingString:@"/amplitude_identify_storage.txt"];
+    return path;
+}
+
++ (void)storeEvent:(NSString *) event {
+    NSString *path = [AMPStorage getDefaultEventsFile];
+    NSURL *url = [NSURL fileURLWithPath:path];
+    [AMPStorage storeEventAtUrl:url event:event];
+}
+
++ (void)storeIdentify:(NSString *) event {
+    NSString *path = [AMPStorage getDefaultIdentifyFile];
+    NSURL *url = [NSURL fileURLWithPath:path];
+    [AMPStorage storeEventAtUrl:url event:event];
+}
+
++ (void)storeEventAtUrl:(NSURL *)url event:(NSString *)event {
```

Comment:
Thinking about the corner case here. Is it possible that we call `finish`, but have events stuck in file storage and we store more events in the files later after this?
Let's say if we call `storeEventAtUrl`, then the app enters the background and call `finish` to close the file. But if we cannot send events to the server successfully, we will not remove the storage files and the events stuck in file storage. But if we go back to the application, and do actions we are still able to logEvent. But since we append `@"]}";` at the end of the file, if we keep writing events, the format will not right. 
Don't know if I'm overthinking this.

Reply:
If all the operations are queued into the same queue to execute, then it should be fine. We need to guarantee that.

Reply:
I set up `finish` only to be called after the app goes into the background. I thought about this case a lot.

So the SDK starts up, we write events to it until we send. The file will be incomplete at this point (no ending braces). 

Events are then sent. It will be a success, which means we delete the file. Or it will fail, and we keep the file open (still no ending braces).

Then, the braces are added when the app enters the background. This is so they can be retrieved later as JSON.